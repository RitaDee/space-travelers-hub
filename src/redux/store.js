import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './mission/MissionSlice';

const store = configureStore({
  reducer: {
    missions: missionReducer,
  },
});

export default store;
