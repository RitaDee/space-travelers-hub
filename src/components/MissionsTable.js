import React from 'react';
import { useSelector } from 'react-redux';

import styles from 'src/css/MissionsTable.module.css';
import { selectMissions } from 'src/redux/missionsSlice';

import MissionsLabel from './MissionsLabel';
import MissionsButton from './MissionsButton';

function MissionsTable() {
  const missions = useSelector(selectMissions);

  return (
    <table className={styles.table}>
      <colgroup>
        <col className={styles.col1} />
        <col className={styles.col2} />
        <col className={styles.col3} />
        <col className={styles.col4} />
      </colgroup>
      <thead>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th> </th>
        </tr>
      </thead>
      <tbody>
        {missions.map((mission) => (
          <tr key={mission.mission_id} className={styles.missionRow}>
            <td className={styles.name}>{mission.mission_name}</td>
            <td>
              <p className={styles.description}>{mission.description}</p>
            </td>
            <td className={styles.center}>
              <MissionsLabel isReserved={!!mission.reserved} />
            </td>
            <td className={styles.center}>
              <MissionsButton isReserved={!!mission.reserved} />
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  );
}

export default MissionsTable;
