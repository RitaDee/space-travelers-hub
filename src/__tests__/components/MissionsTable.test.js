import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import MissionsTable from '../../components/MissionsTable';
import store from '../../redux/store';

describe('MissionsTable', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MissionsTable />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
