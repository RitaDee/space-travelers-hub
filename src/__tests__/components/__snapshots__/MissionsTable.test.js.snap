// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionsTable renders correctly 1`] = `
<table
  className="table"
>
  <colgroup>
    <col
      className="col1"
    />
    <col
      className="col2"
    />
    <col
      className="col3"
    />
    <col
      className="col4"
    />
  </colgroup>
  <thead>
    <tr>
      <th>
        Mission
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
      <th>
         
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
